lib/membrane/video_compositor/scene/object/layout.ex

```diff
@@ -10,11 +10,62 @@ defmodule Membrane.VideoCompositor.Scene.Object.Layout do
   """
   alias Membrane.VideoCompositor.Scene.{Object, Resolution}
 
+  defmodule RustlerFriendly do
+    @moduledoc false
+    # A rustler-friendly version of the Layout, prepared for rust serialization
+    alias Membrane.VideoCompositor.Scene.Object.RustlerFriendly, as: RFObject
+    alias Membrane.VideoCompositor.Scene.Resolution
+
+    @typedoc """
+    A rustler-friendly version of the internal name
+    """
+    @type internal_name :: RFObject.name()
+
+    @typedoc """
+    A rustler-friendly version of the inputs map. Keep in mind that this maps
+    __internal names__ => __scene object names__.
+    """
+    @type inputs :: %{internal_name() => RFObject.name()}
+    @type output_resolution :: {:resolution, Resolution.t()} | {:name, Object.name()}
+
+    # in a more 'final' product this should be some kind of a layout identifier.
+    # I thought of making this a UUID that would correspond to an implementation
+    # on the rust side, but layout names would work fine too.
+    @type rust_representation :: integer()
+
+    @type t :: %__MODULE__{
+            :inputs => inputs(),
+            :resolution => output_resolution(),
+            # unsure about calling this `implementation`.
+            :implementation => rust_representation()
+          }
+
+    @enforce_keys [:inputs, :resolution, :implementation]
+    defstruct @enforce_keys
+  end
+
   @typedoc """
-  Type of a map defining on how to map internal layout's ids
-  to Scene objects
+  A layout-internal identifier, which the layout can use to determine
+  the way of rendering specific input objects. See `t:inputs/0` for more info.
   """
-  @type inputs :: %{any() => Object.input()}
+  @type internal_name :: Object.name()
```

Comment:
In some cases `atom` in `{atom(), number}` tuple feels unnecessary. e.g. repeating `:input` in your tests
```suggestion
  @type internal_name :: Object.name() | integer() | float()
```